File: files/samples/Scroller.js

//Called when application is started.
function OnStart()
{
    //Create the main layout.
    lay = app.CreateLayout( "linear", "FillXY" )    
    
    //Create a full screen scroller
    scroll = app.CreateScroller( 1.0, 1.0 )
    lay.AddChild( scroll )
 
    //Create a layout inside scroller.
    layScroll = app.CreateLayout( "Linear", "Left" )
    scroll.AddChild( layScroll )
    
    //Create an image twice the screen size.
    img = app.CreateImage( "/Sys/Img/Hello.png", 2.0, 2.0 )
    layScroll.AddChild( img )
       
    //Add layout to app.    
    app.AddLayout( lay )
    
    //Initially scroll to center of image.
    scroll.ScrollTo( 0.5, 0.5 )
    
    //Show the current scroll position every second.
    setInterval( ShowScrollPos, 1000 )
}

//Called every second.
function ShowScrollPos()
{
    //Current scroll position and diplay it.
    var x = scroll.GetScrollX()
    var y = scroll.GetScrollY()
    app.ShowPopup( x.toFixed(2) + ", " + y.toFixed(2) )
}

File: files/samples/Scroller.py
from native import app

def OnStart():
    global scroll

    lay = app.CreateLayout("linear", "FillXY")

    scroll = app.CreateScroller(1.0, 1.0)
    lay.AddChild(scroll)

    layScroll = app.CreateLayout("Linear", "Left")
    scroll.AddChild(layScroll)

    img = app.CreateImage("/Sys/Img/Hello.png", 2.0, 2.0)
    layScroll.AddChild(img)

    app.AddLayout(lay)

    scroll.ScrollTo(0.5, 0.5)

    setInterval(ShowScrollPos, 1000)

def ShowScrollPos():
    x = scroll.GetScrollX()
    y = scroll.GetScrollY()
    app.ShowPopup(str(round(x, 2)) + ", " + str(round(y, 2)))